content/projects/gsoc/2019/artifact-promotion-plugin-for-jenkins-pipeline.adoc

```diff
@@ -33,8 +34,9 @@ with other artifact tracking and job triggering functions.
 Sub-tasks to complete the project:
 
 
-. Refactor the selected modules inside the promoted-builds-plugin and make them pipeline compatible.
-. Detach all the required extension points and modules into the new plugin.
+. Refactor the selected extension points(if possible)inside the promoted-builds-plugin and make them pipeline compatible.
+. If the extension points seem to be tightly coupled and look to often break the API/binary compatibilty then create new classes for the same.
+. For core modules(eg. PromotionProcess) opt to create new analogous classes within the promoted-builds-plugin.
```

Comment:
Most readers don't remember the phrase "exempli gratia" or its abbreviation "e.g.".  Simpler to use the phrase "for example".
```suggestion
. For core modules (for example, PromotionProcess) opt to create new analogous classes within the promoted-builds-plugin.
```